tools/clang/lib/SPIRV/EmitSPIRVAction.cpp

```diff
@@ -675,93 +671,81 @@ class SPIRVEmitter : public ASTConsumer {
     // First handle the condition variable DeclStmt if one exists.
     // For example: handle 'int a = b' in the following:
     // switch (int a = b) {...}
-    const auto *condVarDeclStmt = switchStmt->getConditionVariableDeclStmt();
-    if (condVarDeclStmt)
+    if (const auto *condVarDeclStmt =
+            switchStmt->getConditionVariableDeclStmt())
       doStmt(condVarDeclStmt);
 
-    /// Figure out the indexes of CaseStmts (and DefaultStmt if it exists) in
-    /// the flattened switch AST.
-    /// For instance, for the following flat vector, the indexes are:
-    /// {0, 2, 5, 6, 9}
-    /// +-------------------------------------------------------------------+
-    /// |Case1|Stmt1|Case2|Stmt2|Break|Case3|Case4|Stmt4|Break|Default|Stmt5|
-    /// +-------------------------------------------------------------------+
+    // Figure out the indexes of CaseStmts (and DefaultStmt if it exists) in
+    // the flattened switch AST.
+    // For instance, for the following flat vector, the indexes are:
+    // {0, 2, 5, 6, 9}
+    // +-------------------------------------------------------------------+
+    // |Case1|Stmt1|Case2|Stmt2|Break|Case3|Case4|Stmt4|Break|Default|Stmt5|
+    // +-------------------------------------------------------------------+
     std::vector<size_t> caseStmtLocs;
-    for (int i = 0; i < flatSwitch.size(); ++i)
-      if (dyn_cast<CaseStmt>(flatSwitch[i]) ||
-          dyn_cast<DefaultStmt>(flatSwitch[i]))
+    for (uint32_t i = 0; i < flatSwitch.size(); ++i)
+      if (isa<CaseStmt>(flatSwitch[i]) || isa<DefaultStmt>(flatSwitch[i]))
         caseStmtLocs.push_back(i);
 
-    std::vector<std::shared_ptr<BinaryOperator>> comparisons;
-    std::vector<std::shared_ptr<CompoundStmt>> bodies;
-    std::shared_ptr<CompoundStmt> defaultBody;
-    std::vector<IfStmt> ifs;
+    std::vector<IfStmt *> ifs;
```

Comment:
It seems we actually only need to keep track of the previous `if` statement generated using like `prevIfStmt`. Init `prevIfStmt` to `nullptr` and then in the `for` loop:
```c++
for (...) {
  ...
  if (prevIfStmt)  prevIfStmt->setElse(curIf);
  prevIfStmt = curIf;
}

if (prevIfStmt) prevIfStmt->setElse(defaultBody);
```